packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -638,71 +1050,133 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
+  Color? _cachedThumbColor;
+  ImageProvider? _cachedThumbImage;
+  ImageErrorListener? _cachedThumbErrorListener;
+  BoxPainter? _cachedThumbPainter;
+
+  ShapeDecoration _createDefaultThumbDecoration(Color color, ImageProvider? image, ImageErrorListener? errorListener) {
+    return ShapeDecoration(
+      color: color,
+      image: image == null ? null : DecorationImage(image: image, onError: errorListener),
+      shape: const StadiumBorder(),
+    );
   }
 
-  @override
-  void describeSemanticsConfiguration(SemanticsConfiguration config) {
-    super.describeSemanticsConfiguration(config);
+  bool _isPainting = false;
 
-    if (isInteractive) {
-      config.onTap = _state._handleTap;
+  void _handleDecorationChanged() {
+    // If the image decoration is available synchronously, we'll get called here
+    // during paint. There's no reason to mark ourselves as needing paint if we
+    // are already in the middle of painting. (In fact, doing so would trigger
+    // an assert).
+    if (!_isPainting) {
+      notifyListeners();
     }
-
-    config.isEnabled = isInteractive;
-    config.isToggled = _value;
   }
 
-  @override
-  void paint(PaintingContext context, Offset offset) {
-    final Canvas canvas = context.canvas;
+  bool _stopPressAnimation = false;
+  double? _pressedInactiveThumbRadius;
+  double? _pressedActiveThumbRadius;
+  late double? _pressedThumbExtension;
 
-    final double currentValue = _state.position.value;
-    final double currentReactionValue = _state._reaction.value;
+  @override
+  void paint(Canvas canvas, Size size) {
+    final double currentValue = position.value;
 
     final double visualPosition = switch (textDirection) {
       TextDirection.rtl => 1.0 - currentValue,
       TextDirection.ltr => currentValue,
     };
+    if (reaction.status == AnimationStatus.reverse && !_stopPressAnimation) {
+      _stopPressAnimation = true;
+    } else {
+      _stopPressAnimation = false;
+    }
+
+    // To get the thumb radius when the press ends, the value can be any number
+    // between activeThumbRadius/inactiveThumbRadius and pressedThumbRadius.
+    if (!_stopPressAnimation) {
+      _pressedThumbExtension = reaction.value * 7;
+      if (reaction.isCompleted) {
+        // This happens when the thumb is dragged instead of being tapped.
+        _pressedInactiveThumbRadius = lerpDouble(inactiveThumbRadius, pressedThumbRadius, reaction.value);
+        _pressedActiveThumbRadius = lerpDouble(activeThumbRadius, pressedThumbRadius, reaction.value);
+      }
+      if (currentValue == 0) {
+        _pressedInactiveThumbRadius = lerpDouble(inactiveThumbRadius, pressedThumbRadius, reaction.value);
+        _pressedActiveThumbRadius = activeThumbRadius;
+      }
+      if (currentValue == 1) {
+        _pressedActiveThumbRadius = lerpDouble(activeThumbRadius, pressedThumbRadius, reaction.value);
+        _pressedInactiveThumbRadius = inactiveThumbRadius;
+      }
```

Comment:
Just one more question(sorry), will we move this `_SwitchPainter ` class in base widget so material and cupertino can use for both Switch? The reason I'm asking is that this block of code is for the pressing behavior in Material Switch. The thumb of the material switch has several different radius in different states, like when we press material switch, the thumb becomes larger, and inactive radius is also different from the active radius. But for Cupertino switch, seems we only need to care about the thumb extension which is handled below.

If we only want to focus on Cupertino here for now, maybe we can just remove these material-related code to make the implementation simpler. But if we eventually want to move this part to a base widget which needs to include both behavior, then probably we can keep the same implementation here.

Reply:
I'm not sure to be honest. Although for this PR I think we should remove it anyways just so there isn't dead code floating around here. If we need to add it back it should be trivial (I think).

Reply:
Sounds good! Then I can take another look to double check whether we have some more extra things to remove! (I'll need to double check but I think `activeThumbRadius` and `inactiveThumbRadius` may also be redundant because CupertinoSwitch should just have the same radius when it's on and off)

Reply:
I think they are redundant too. I removed them, along with those other material-related code.